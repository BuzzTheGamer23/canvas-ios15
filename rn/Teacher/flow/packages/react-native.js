// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
