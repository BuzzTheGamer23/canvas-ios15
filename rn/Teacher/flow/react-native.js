//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
