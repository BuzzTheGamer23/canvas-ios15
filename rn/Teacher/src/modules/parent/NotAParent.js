//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import React, { PureComponent } from 'react'
import {
  View,
  StyleSheet,
  Button,
  Image,
  TouchableOpacity,
  NativeModules,
  Linking,
} from 'react-native'
import i18n from 'format-message'
import { Text, Paragraph } from '../../common/text'
import Images from '../../images'
import Screen from '../../routing/Screen'
import colors from '../../common/colors'

export class NotAParent extends PureComponent<{}> {
  logout = () => {
    NativeModules.NativeLogin.logout()
  }

  canvasGuides = () => {
    Linking.openURL('https://community.canvaslms.com/docs/DOC-9919')
  }

  openStudent = () => {
    Linking.openURL('https://itunes.apple.com/us/app/canvas-by-instructure/id480883488?ls=1&mt=8')
  }

  openTeacher = () => {
    Linking.openURL('https://itunes.apple.com/us/app/canvas-teacher/id1257834464?ls=1&mt=8')
  }

  render () {
    let title = i18n('No Enrollments')
    let subtitleOne = i18n('You need at least one active observer enrollment to use Canvas Parent.')
    let subtitleTwo = i18n("If you're a student or teacher, try one of our other apps below.")

    return (
      <Screen navBarHidden>
        <View style={style.container}>
          <View style={style.subContainer}>
            <Text style={style.title} testID='no-parent.title'>{title}</Text>
            <Paragraph style={style.paragraph} testID='no-parent.subtitleOne'>{subtitleOne}</Paragraph>
            <View style={style.separator} />
            <Paragraph style={style.paragraph} testID='no-parent.subtitleTwo'>{subtitleTwo}</Paragraph>
          </View>
          <View style={style.subContainer}>
            <TouchableOpacity onPress={this.openStudent}
              testID='no-parent.open-student'
              accessibilityTraits="button"
              accessibilityLabel={i18n('Canvas Student App')}>
              <Image source={Images.noTeacher.student} />
            </TouchableOpacity>
            <TouchableOpacity style={{ marginTop: 36 }}
              onPress={this.openTeacher} testID='no-parent.open-teacher'
              accessibilityTraits="button"
              accessibilityLabel={i18n('Canvas Teacher App')}>
              <Image source={Images.noTeacher.teacher} />
            </TouchableOpacity>
          </View>
          <View style={[style.subContainer, { marginTop: 50 }]}>
            <Button title={i18n('Log In Again')} testID='no-parent.logout' onPress={this.logout} />
          </View>
          <View style={style.subContainer}>
            <Button title={i18n('Canvas Guides')} testID='no-parent.canvas-guides' onPress={this.canvasGuides} />
          </View>
        </View>
      </Screen>
    )
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
  subContainer: {
    margin: 8,
  },
  title: {
    fontSize: 30,
    textAlign: 'center',
  },
  paragraph: {
    fontSize: 16,
    textAlign: 'center',
    padding: 8,
    maxWidth: 320,
  },
  separator: {
    backgroundColor: colors.seperatorColor,
    height: StyleSheet.hairlineWidth,
    marginTop: 20,
    marginBottom: 20,
  },
})

export default (NotAParent: any)
