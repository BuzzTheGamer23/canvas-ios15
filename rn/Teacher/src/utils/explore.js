/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
