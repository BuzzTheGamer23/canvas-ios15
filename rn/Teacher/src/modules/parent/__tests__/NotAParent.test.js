//
// Copyright (C) 2017-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/**
 * @flow
 */

import { NativeModules, Linking } from 'react-native'
import React from 'react'
import NotAParent from '../NotAParent'
import explore from '../../../../test/helpers/explore'

jest.mock('TouchableOpacity', () => 'TouchableOpacity')

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders NotAParent correctly', () => {
  let tree = renderer.create(
    <NotAParent />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('all the tapping should work', () => {
  let tree = renderer.create(
    <NotAParent />
  ).toJSON()

  const logout = explore(tree).selectByID('no-parent.logout') || {}
  const student = explore(tree).selectByID('no-parent.open-student') || {}
  const parent = explore(tree).selectByID('no-parent.open-teacher') || {}

  logout.props.onPress()
  expect(NativeModules.NativeLogin.logout).toHaveBeenCalled()
  student.props.onPress()
  expect(Linking.openURL).toHaveBeenCalledWith('https://itunes.apple.com/us/app/canvas-by-instructure/id480883488?ls=1&mt=8')
  parent.props.onPress()
  expect(Linking.openURL).toHaveBeenCalledWith('https://itunes.apple.com/us/app/canvas-teacher/id1257834464?ls=1&mt=8')
})
