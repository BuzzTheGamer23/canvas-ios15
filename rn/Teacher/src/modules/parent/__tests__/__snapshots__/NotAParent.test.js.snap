// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotAParent correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "margin": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#2D3B45",
              "fontSize": 16,
            },
            Object {
              "fontSize": 30,
              "textAlign": "center",
            },
            Object {
              "fontFamily": ".SFUIDisplay",
            },
          ]
        }
        testID="no-parent.title"
      >
        No Enrollments
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#7F91A7",
              "fontSize": 16,
              "lineHeight": 23,
            },
            Object {
              "fontSize": 16,
              "maxWidth": 320,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
        testID="no-parent.subtitleOne"
      >
        You need at least one active observer enrollment to use Canvas Parent.
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#C7CDD1",
            "height": 0.5,
            "marginBottom": 20,
            "marginTop": 20,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": ".SFUIDisplay",
            },
            Object {
              "color": "#7F91A7",
              "fontSize": 16,
              "lineHeight": 23,
            },
            Object {
              "fontSize": 16,
              "maxWidth": 320,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
        testID="no-parent.subtitleTwo"
      >
        If you're a student or teacher, try one of our other apps below.
      </Text>
    </View>
    <View
      style={
        Object {
          "margin": 8,
        }
      }
    >
      <TouchableOpacity
        accessibilityLabel="Canvas Student App"
        accessibilityTraits="button"
        onPress={[Function]}
        testID="no-parent.open-student"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/Canvas-Student.png",
            }
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        accessibilityLabel="Canvas Teacher App"
        accessibilityTraits="button"
        onPress={[Function]}
        style={
          Object {
            "marginTop": 36,
          }
        }
        testID="no-parent.open-teacher"
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/Canvas-Teacher.png",
            }
          }
        />
      </TouchableOpacity>
    </View>
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "marginTop": 50,
          },
        ]
      }
    >
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        disabled={undefined}
        hasTVPreferredFocus={undefined}
        onPress={[Function]}
        testID="no-parent.logout"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Log In Again
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      style={
        Object {
          "margin": 8,
        }
      }
    >
      <TouchableOpacity
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        disabled={undefined}
        hasTVPreferredFocus={undefined}
        onPress={[Function]}
        testID="no-parent.canvas-guides"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={undefined}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Canvas Guides
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</RCTSafeAreaView>
`;
